### PR TITLE
python3.pkgs.gcodepy: init at 0.1.1

### PR DESCRIPTION
**Pull request by n00b0ss**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/gcodepy/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  pythonImportsCheck = [
+    "gcodepy"
+  ];
+
+  meta = with lib; {
+    description = "gcodepy is a g-code generator for 3D printers that use Marlin Firmware.";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/gcodepy/default.nix:28]:
```suggestion
    description = "G-code generator for 3D printers that use Marlin Firmware.";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Thank you for packaging gcodepy!

Please reorder your commits so the maintainers change comes first. Your commits are also not [associated with your account](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user).

File: pkgs/development/python-modules/gcodepy/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+buildPythonPackage rec {
+  pname = "gcodepy";
+  version = "0.1.1";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "rmeno12";
+    repo = "gcodepy";
+    rev = "refs/tags/v${version}";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/gcodepy/default.nix:15]:
```suggestion
    rev = "v${version}";
```
This should automatically fetch the tag.

File: pkgs/development/python-modules/gcodepy/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  pythonImportsCheck = [
+    "gcodepy"
+  ];
+
+  meta = with lib; {
+    description = "G-code generator for 3D printers that use Marlin Firmware.";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/gcodepy/default.nix:28]:
```suggestion
    description = "G-code generator for 3D printers that use Marlin Firmware";
```
No period, as specified in the [docs](https://ryantm.github.io/nixpkgs/stdenv/meta/#var-meta-description).

**Review by FliegendeWurst — COMMENTED** [review]
Package LGTM besides two minor points.

You'll need to make sure there are only two commits: "maintainers: add n00b0ss" and the package init. This policy is fairly strict.
It can be done in two ways:
- interactive rebase (see e.g. [here](https://www.jetbrains.com/guide/java/tutorials/git-interactive-rebase/cli/) for a tutorial), or equivalent in a git GUI wrapper
- create a new branch and [copy the changes over manually](https://xkcd.com/1597/), then open a new PR or force-push that branch to `gcodepy`

**Comment by n00b0ss** [comment]
done

**Review by FliegendeWurst — APPROVED** [review]
Result of `nixpkgs-review pr 259444` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.gcodepy</li>
    <li>python310Packages.gcodepy.dist</li>
    <li>python311Packages.gcodepy</li>
    <li>python311Packages.gcodepy.dist</li>
  </ul>
</details>

**Comment by emilytrau** [comment]
Welcome to nixpkgs! Thanks for your contribution

**Pull request merged**